crates/extensions/tedge_health_ext/src/lib.rs

```diff
@@ -53,10 +113,11 @@ impl HealthMonitorBuilder {
     }
 
     fn set_init_and_last_will(&self, config: MqttConfig) -> MqttConfig {
-        let name = self.service_name.to_owned();
+        let name = self.service_health_topic.to_owned();
+        let _name = name.clone();
         config
-            .with_initial_message(move || health_status_up_message(&name))
-            .with_last_will_message(health_status_down_message(&self.service_name))
+            .with_initial_message(move || _name.up_message())
```

Comment:
Surprised that clippy didn't complain here about using a variable with `_` prefix.